contracts/AssetHolder.sol

```diff
@@ -0,0 +1,210 @@
+pragma solidity ^0.5.11;
+pragma experimental ABIEncoderV2;
+import './Outcome.sol';
+import 'openzeppelin-solidity/contracts/math/SafeMath.sol';
+
+contract AssetHolder {
+    using SafeMath for uint256;
+
+    struct Authorization {
+        // Prevents replay attacks:
+        // It's required that the participant signs the message, meaning only
+        // the participant can authorize a withdrawal.
+        // Moreover, the participant should sign the address that they wish
+        // to send the transaction from, preventing any replay attack.
+        address participant; // the account used to sign commitment transitions
+        bytes32 destination; // either an account or a channel
+        uint256 amount;
+        address sender; // the account used to sign transactions
+    }
```

Comment:
Is this still required?

Reply:
I think we probably still need the `participant`'s signature on something. But we may not need to include the `sender` in the message hash, if `withdraw` is to be an internal call from `transferAll`.